cmd/clusterctl/client/tree/discovery.go

```diff
@@ -69,9 +70,13 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-	if err == nil {
-		tree.Add(cluster, controlPLane, ObjectMetaName("ControlPlane"), GroupingObject(true))
+	var controlPlane *unstructured.Unstructured
+	if cluster.Spec.ControlPlaneRef != nil {
+		var err error
+		controlPlane, err = external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
+		if err == nil {
```

Comment:
Seems this error is effectively being ignored now?

Reply:
+1 to return an error here given that we are reviewing this code